### PR TITLE
feat: ability to update used html template when an element is selected

### PR DESCRIPTION
**Pull request by Sparksx**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -93,9 +93,19 @@
               console.log('shouldShowInformationButtonFor', e)
               return '<span class="mwz-icon-information">i</span> ' + e.name
             },
-            // onElementWillBeSelected: function (e) {
-            //   return { pitch: 50, bearing: 50, zoom: 20, centerOnElement: true }
-            // },
+            onElementWillBeSelected: function (element, options) {
+              // return { pitch: 50, bearing: 50, zoom: 20, centerOnElement: true } // NEVER DO THAT, IT WILL BREAK NEXT RELEASES
```
Comment (mathieugerard) [inline src/index.html:97]: Why is this not allowed? Options should be defaulted.

File: tests/events/on-element-will-be-selected.test.js [hunk 2]
```diff
@@ -7,8 +7,13 @@ mwzDescribe(testSuites, function () {
...
       centerOnPlaceId: MAPWIZEPLACEID,
-      onElementWillBeSelected: function (e) {
-        return { pitch: 53, bearing: 50, zoom: 20, centerOnElement: true }
+      onElementWillBeSelected: function (element, options) {
+        // return { pitch: 53, bearing: 50, zoom: 20, centerOnElement: true } // NEVER DO THAT, IT WILL BREAK NEXT RELEASES
+        options.pitch = 53
+        options.bearing = 50
+        options.zoom = 20
+        options.centerOnElement = true;
+        return options
```
Comment (mathieugerard) [inline tests/events/on-element-will-be-selected.test.js:16]: Would be smart to have a simple test to make sure the template is used.

File: README-advanced.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+The function should return `options`.
+
+### onDirectionQueryWillBeSent
+
+`onDirectionQueryWillBeSent` is called before a direction request is sent. The interceptor is a `function (query)` that should return `query`.
+
+### onDirectionWillBeDisplayed
+
+`onDirectionWillBeDisplayed` is called before a direction is displayed. The interceptor is a `function (direction, options)` where `direction` is the direction object to be displayed and the `options` defines the changes in the UI. The fonction should return `options`.
```
Comment (Sparksx) [inline README-advanced.md:32]: should return both `direction` and `options`
`function (direction, options) {  return { direction: direction, options: options }; }`

File: README-advanced.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+```
+function (results) { return results; },
+```
+
+## Direction mode icons
+
+This is the procedure to change the icons of the direction modes
+
+- Open svg with edior
```
Comment (Sparksx) [inline README-advanced.md:50]:
```suggestion
- Open svg with text editor (like `sublime text`)
```

**Pull request merged**